packages/sdk-core/src/proving/proving-manager-thread.ts

```diff
@@ -252,10 +255,21 @@ export class ProvingManagerWrapper {
       pm.setRoots(input.roots);
       pm.chain_id(input.chainId);
       pm.public_amount(input.publicAmount);
-      pm.setVanchorOutputConfig(...input.outputConfigs);
-      pm.setExtDatahash(input.externalDataHash);
-      // leaves insertion
+      const [o1, o2] = pm.setVanchorOutputConfig(...input.outputConfigs) as [JsUtxo, JsUtxo];
       const wasm = await this.wasmBlob;
+      const extData = new wasm.ExtData(
+        input.recipient,
+        input.relayer,
+        input.extAmount,
+        input.fee,
+        o1.commitment,
```

Comment:
This should be an encryption from polkadotjs to start, no? https://polkadot.js.org/docs/util-crypto/examples/encrypt-decrypt/